armi/physics/fuelCycle/fuelHandlers.py

```diff
@@ -997,14 +1104,133 @@ def readMoves(fname):
+                runLog.warning(f"Cycle {cycleKey} has no shuffle actions defined, skipping.")
+                continue
+
+            elif cycle == 0:
+                raise InputError(
+                    "Cycle 0 is not allowed in shuffle YAML. "
+                    "This cycle is reserved for the initial core loading."
+                    "Shuffling is available at the beginning of cycle 1"
+                )
+
+            for action in actions:
+                allowed = {"cascade", "fuelEnrichment", "extraRotations", "misloadSwap"}
+                unknown = set(action) - allowed
+                if unknown:
+                    raise InputError(f"Unknown action keys {unknown} in shuffle YAML")
+
+                if "cascade" in action:
+                    chain = list(action["cascade"])
+                    if len(chain) < 2:
+                        raise InputError("cascade must contain an assembly type and at least one location")
+                    if any(not isinstance(item, str) for item in chain):
+                        raise InputError("cascade entries must be strings")
+
+                    assemType = chain[0]
+                    locs = chain[1:]
+                    for loc in locs:
+                        FuelHandler._validateLoc(loc, cycle)
+                        if loc not in FuelHandler.DISCHARGE_LOCS and loc in seenLocs:
+                            raise InputError(f"Location {loc} appears in multiple cascades in cycle {cycle}")
+                        seenLocs.add(loc)
+
+                    enrich = []
+                    enrichList = action.get("fuelEnrichment", [])
+                    try:
+                        enrich = [float(e) for e in enrichList]
+                    except (TypeError, ValueError):
+                        raise InputError("fuelEnrichment values must be numeric. Got {enrichList}")
+                    if any(e < 0 or e > 1 for e in enrich):
+                        raise InputError("fuelEnrichment values must be between 0 and 1. Got {enrich}")
+
+                    moves[cycle].append(AssemblyMove("LoadQueue", locs[0], enrich, assemType))
+                    for i in range(len(locs) - 1):
+                        moves[cycle].append(AssemblyMove(locs[i], locs[i + 1]))
+                    if locs[-1] not in FuelHandler.DISCHARGE_LOCS:
+                        moves[cycle].append(AssemblyMove(locs[-1], "SFP"))
+
+                elif "misloadSwap" in action:
+                    swap = action["misloadSwap"]
+                    if not isinstance(swap, list) or len(swap) != 2:
+                        raise InputError("misloadSwap must be a list of two location labels, got {swap}")
+                    if any(not isinstance(item, str) for item in swap):
+                        raise InputError("misloadSwap entries must be strings, got {swap}")
+                    for loc in swap:
+                        FuelHandler._validateLoc(loc, cycle)
```

Comment:
We are accessing `_validateLoc()` outside the FuelHandler? Than this should be a public state method, not a private method.

Can we change that? You added that in this PR, right?